### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by nico-nico13**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    public void selectRadio() {
+        selen.click();
+    }
+
+    public void selectDropdown() {
+        Select select = new Select(dropdown);
+        select.selectByIndex(3);
+    }
+
+    public String waterLogTest() {
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java:51]: Please replace `Test` to `Text` for all page objects methods that return element text

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public void selectCheckboxex() {
+        water.click();
+        wind.click();
+    }
+
+    public void selectRadio() {
+        selen.click();
+    }
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java:44]: This page contains several radios. But the method `selectRadio` always perform click on selen radio.
I would replace with `selectRadio(String radioName)` or `selectSelenRadio()`

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    private WebElement colorsLog;
+
+    public DifferentElementsPage(WebDriver driver) {
+        PageFactory.initElements(driver, this);
+    }
+
+    public void selectCheckboxex() {
+        water.click();
+        wind.click();
+    }
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java:40]: This page contains several checkboxes. But the method `selectCheckboxex()` always perform clicks on water and wind.
I would replace with `selectCheckbox(String checkboxName)` or `selectWaterCheckbox()` and `selectWindCheckbox()`

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    public String waterLogTest() {
+        String waterLogText = waterLog.getText();
+        return waterLogText;
+    }
+
+    public String windLogTest() {
+        String windLogText = windLog.getText();
+        return windLogText;
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java:58]: can be simplified with
```java
return windLog.getText();
```

File: src/test/java/com/epam/tc/hw3/pages/IndexPage.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    }
+
+    public boolean headerMetalsColorsDisplay() {
+        return headerMetalsColors.isDisplayed();
+    }
+
+    public boolean headerHomeTextTest() {
+        String headerHomeText = headerHome.getText();
+        return headerHomeText.equals("HOME");
+    }
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw3/pages/IndexPage.java:106]: Please move test logic and test data out of page objects

File: src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+
+public class FirstExercise extends BaseClass {
+
+    SoftAssert softAssert = new SoftAssert();
+
+    @Test
+    public void firstTest() {
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java:14]: Please rename test method with some meaningful name.
Test methods should have a name in accordance with checking functionality

File: src/test/java/com/epam/tc/hw3/ex2/SecondExercise.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import com.epam.tc.hw3.pages.IndexPage;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+public class SecondExercise extends BaseClass {
+
+    SoftAssert softAssert = new SoftAssert();
+
+    @Test
+    public void secondTest() {
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw3/ex2/SecondExercise.java:14]: Please rename test method with some meaningful name.
Test methods should have a name in accordance with checking functionality

File: src/test/resources/AllTests.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    <test verbose="2" preserve-order="true" name="AllTests">
+        <classes>
+            <class name="com.epam.tc.hw3.ex1.FirstExercise">
+                <methods>
+                    <include name="FirstTest"/>
+                </methods>
+            </class>
+            <class name="com.epam.tc.hw3.ex2.SecondExercise">
+                <methods>
+                    <include name="Second"/>
```
Comment (akulebyakin) [inline src/test/resources/AllTests.xml:13]: Don't forget to change xml suite after you rename test methods according to requirement
```
Test methods should have a name in accordance with checking functionality
```

**Review by akulebyakin — CHANGES_REQUESTED** [review]
Please review comments

File: src/test/java/com/epam/tc/hw3/ex2/SecondExercise.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        IndexPage indexPage = new IndexPage(driver);
+        // 2. Assert Browser title
+        softAssert.assertEquals(driver.getTitle(), "Home Page");
+        // 3. Perform login
+        indexPage.login("Roman", "Jdi1234");
```
[the remark this answers is not in the document]
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/ex2/SecondExercise.java:19]: Done

File: src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        softAssert.assertTrue(indexPage.text2Test());
+        softAssert.assertTrue(indexPage.text3Test());
+        softAssert.assertTrue(indexPage.text4Test());
+        // 8. Assert that there is the iframe with “Frame Button” exists
+        softAssert.assertTrue(indexPage.iframeEnable());
+        // 9.
+        driver.switchTo().frame("frame");
+        softAssert.assertTrue(indexPage.frameButtonEnable());
+        // 10.
+        driver.switchTo().defaultContent();;
```
[the remark this answers is not in the document]
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java:48]: Done

File: src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+import com.epam.tc.hw3.BaseClass;
+import com.epam.tc.hw3.pages.IndexPage;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+
+public class FirstExercise extends BaseClass {
+
+    SoftAssert softAssert = new SoftAssert();
```
[the remark this answers is not in the document]
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java:11]: Added assertAll and initialization moved

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java (hunk 1 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java:51]: Replaced

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java (hunk 4 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java:58]: Simplified

File: src/test/java/com/epam/tc/hw3/pages/IndexPage.java (hunk 5 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/pages/IndexPage.java:106]: Moved

File: src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java (hunk 6 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/ex1/FirstExercise.java:14]: Renamed

File: src/test/resources/AllTests.xml (hunk 8 above)
Comment (nico-nico13) [reply src/test/resources/AllTests.xml:13]: Renamed

File: src/test/java/com/epam/tc/hw3/ex2/SecondExercise.java (hunk 7 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw3/ex2/SecondExercise.java:14]: Renamed

**Review by akulebyakin — APPROVED** [review]
approved

**Pull request merged**